x-pack/test/security_solution_cypress/cypress/tasks/api_calls/common.ts

```diff
@@ -115,71 +116,25 @@ export const deleteTimelines = () => {
   });
 };
 
-export const deleteAllCasesItems = () => {
-  const kibanaIndexUrl = `${Cypress.env('ELASTICSEARCH_URL')}/.kibana_alerting_cases_\*`;
-  rootRequest({
-    method: 'POST',
-    url: `${kibanaIndexUrl}/_delete_by_query?conflicts=proceed&refresh`,
-    body: {
-      query: {
-        bool: {
-          filter: [
-            {
-              bool: {
-                should: [
-                  {
-                    term: {
-                      type: 'cases',
-                    },
-                  },
-                  {
-                    term: {
-                      type: 'cases-configure',
-                    },
-                  },
-                  {
-                    term: {
-                      type: 'cases-comments',
-                    },
-                  },
-                  {
-                    term: {
-                      type: 'cases-user-action',
-                    },
-                  },
-                  {
-                    term: {
-                      type: 'cases-connector-mappings',
-                    },
-                  },
-                ],
-              },
-            },
-          ],
-        },
-      },
-    },
+export const getConnectors = () =>
+  rootRequest<AllConnectorsResponse[]>({
+    method: 'GET',
+    url: 'api/actions/connectors',
   });
-};
 
-export const deleteConnectors = () => {
-  const kibanaIndexUrl = `${Cypress.env('ELASTICSEARCH_URL')}/.kibana_alerting_cases_\*`;
-  rootRequest({
-    method: 'POST',
-    url: `${kibanaIndexUrl}/_delete_by_query?conflicts=proceed&refresh`,
-    body: {
-      query: {
-        bool: {
-          filter: [
-            {
-              match: {
-                type: 'action',
-              },
-            },
-          ],
-        },
-      },
-    },
+export const deleteConnectors = async () => {
+  getConnectors().then(($response) => {
+    if ($response.body.length > 0) {
+      const ids = $response.body.map((connector) => {
+        return connector.id;
+      });
+      ids.forEach((id) => {
+        rootRequest({
+          method: 'DELETE',
+          url: `api/actions/connector/${id}`,
+        });
+      });
+    }
```

Comment:
This triggers multiple promises without any await. We should use Promise.all, so if the caller of `deleteConnectors` is awaiting, we ensure the requests are finished:

```suggestion
export const deleteConnectors = async (): Promise<void> => {
  const connectorsResponse = await getConnectors();
  await Promise.all(
    connectorsResponse.body.map(({id}) => rootRequest({
        method: 'DELETE',
        url: `api/actions/connector/${id}`,
      })
    )
  );
```